src/misc/api_trace.cc

```diff
@@ -0,0 +1,418 @@
+
+#include "nccl.h"
+#include <vector>
+#include <iostream>
+#include "api_trace.h"
+#include "core.h"
+#if defined(RCCL_ROCPROFILER_REGISTER) && RCCL_ROCPROFILER_REGISTER > 0
+    #include <rocprofiler-register/rocprofiler-register.h>
+
+#define ROCP_REG_VERSION                                                                           \
+    ROCPROFILER_REGISTER_COMPUTE_VERSION_3(                                                        \
+        RCCL_VERSION_MAJOR, RCCL_VERSION_MINOR, RCCL_VERSION_PATCH)
+
+ROCPROFILER_REGISTER_DEFINE_IMPORT(rccl, ROCP_REG_VERSION)
+#endif
+
+
+namespace rccl {
+
```

Comment:
```suggestion
namespace {
```
Use an anonymous namespace for all impl details of this file